helm-org.el

```diff
@@ -273,6 +273,73 @@ current heading."
         :truncate-lines helm-org-truncate-lines
         :buffer "*helm org capture templates*"))
 
+;;; Org tag completion
+
+;; Based on code from Anders Johansson posted on 3 Mar 2016 at
+;; <https://groups.google.com/d/msg/emacs-helm/tA6cn6TUdRY/G1S3TIdzBwAJ>
+
+;; FIXME: This function fails with a "Wrong number of arguments" error
+;; if its name starts with "helm"?!  If its name starts with anything
+;; else, it works fine!
+(defun not-helm-org-completing-read-tags (prompt coll pred req initial hist def inh)
+  (if (not (string= "Tags: " prompt))
+      ;; Not a tags prompt.  Use normal completion by calling
+      ;; `org-icompleting-read' again without this function in
+      ;; `helm-completing-read-handlers-alist'
+      (let ((helm-completing-read-handlers-alist (rassq-delete-all
```

Comment:
Sorry, I'm not fluent in English.

`rassq-delete-all` function can modify `helm-completing-read-handlers-alist`.
I think commiter have intended temporary changing `helm-completing-read-handlers-alist` in `let` binding.

> https://www.gnu.org/software/emacs/manual/html_node/elisp/Association-Lists.html
> — Function: rassq-delete-all value alist
> This function deletes from alist all the elements whose cdr is eq to value. It returns the shortened alist, and **often modifies the original list structure of alist**. rassq-delete-all is like assq-delete-all except that it compares the cdr of each alist association instead of the car.

1. `org-set-tags` (working) 
2. `org-do-capture` (`helm-completing-read-handlers-alist` changed) 
3. `org-set-tags` (not working)

I tested as following:
```emacs-lisp
(advice-add #'helm-org-completing-read-tags
            :around (lambda (f &rest args)
                      (cl-flet ((rassq-delete-all
                                 (val alist)
                                 (--remove (eq (cdr it) val) alist)))
                        (apply f args))))
```